### DISCUSSION TITLE
Bypassing $:/Import for new tiddlers

### DISCUSSION THREAD

**Discussion by gMakaan** [TiddlyWiki/TiddlyWiki5#7833] [Q&A] [answered]
Hi there, I'm thinking about fast content import in to single file TW5, just drag-n-dropping (or loading with other standard methods) without appearing `$:/Import` - only if no title conflicts detected. Is there some tag/title hack or need to modify import mechanism?

_-MK-_

**Comment by pmario** [comment, marked as answer]
There is no way to do that, with the existing TW UI. 

I personally would not want a possibility that you describe for security reasons. 

TW makes it really easy to import content from other wiki dot-HTML files, from JSON files and so on. 

Without the preview the wiki would import everything, without the user knowing what actually is imported. There could be a lot of "evil" content that you do not really want to land in your wiki.

**Comment by pmario** [comment]
@gMakaan There has been a discussion at Talk TiddlyWiki about adding a data-store to an existing wiki.html file, if the user can control the creation of the data-store file. 

It's possible to use a file-system copy eg: `copy data.store+empty.html new-wiki.html` to create a new wiki that contains all the tiddlers in the store, that was created by a user, without TW. 

Have a closer look at: https://talk.tiddlywiki.org/t/searching-json-records/8184/9?u=pmario it may be a possibility, if you can control the creation of the data store. 

Related info about TW storage structure: https://talk.tiddlywiki.org/t/how-do-tws-json-formats-look-like/5998

**Comment by gMakaan** [comment]
Thanks for explanation! Just trying this for fast import mode for using TW as separate `.tid` file previewer and editor - just drag&drop it on TW opened in browser without save functionality - view or edit it - export as `.tid` back to system. 

I prefer wikitext over markdown for this task if i need more power (transclusions/macros/styling/etc...) and single powerful instrument to work and view with (TW) across all systems (desktop/mobile).
